src/firstboot/firstboot.c

```diff
@@ -748,13 +788,35 @@ static int process_root_password(void) {
         if (lock < 0)
                 return log_error_errno(lock, "Failed to take a lock: %m");
 
-        if (arg_delete_root_password) {
+        if (arg_copy_root_shell && arg_root) {
+                struct passwd *p;
+
+                p = getpwnam("root");
+                if (!p && errno != ENOENT) {
+                        if (!errno)
+                                errno = EIO;
+
+                        return log_error_errno(errno, "Failed to find passwd entry for root: %m");
```

Comment:
`return log_error_errno(errno_or_else(EIO), "Failed to find passwd entry for root: %m");`